src/pages/signing_official_console/SigningOfficialTable2.jsx

```diff
@@ -0,0 +1,286 @@
+  email: {label: 'Email', cellStyle: {width: styles.cellWidths.email}},
+  name: {label: 'Name', cellStyle: {width: styles.cellWidths.name}},
+  libraryCard: {label: 'Library Card', cellStyle: {width: styles.cellWidths.libraryCard}},
+  role: {label: 'Role', cellStyle: {width: styles.cellWidths.role}},
+};
+
+const researcherFilterFunction = getSearchFilterFunctions().signingOfficialResearchers;
+
+const handleClick = async (researcher, specificDac, filteredDaas, checked, refreshResearchers, setResearchers) => {
+  if (!checked) {
+    try {
+      const daaId = filteredDaas.find(daa => daa.dacs.some(dac => dac.dacId === specificDac.dacId))?.daaId;
+      await DAA.createDaaLcLink(daaId, researcher.userId);
+      Notifications.showSuccess({text: `Approved access to ${specificDac.name} to user: ${researcher.displayName}`});
+      refreshResearchers(setResearchers);
+    } catch(error) {
+      Notifications.showError({text: `Error approving access to ${specificDac.name} to user: ${researcher.displayName}`});
+    }
+  } else {
+    try {
+      const daaId = filteredDaas.find(daa => daa.dacs.some(dac => dac.dacId === specificDac.dacId))?.daaId;
+      await DAA.deleteDaaLcLink(daaId, researcher.userId);
+      Notifications.showSuccess({text: `Removed approval of access to ${specificDac.name} to user: ${researcher.displayName}`});
+      refreshResearchers(setResearchers);
+    } catch(error) {
+      Notifications.showError({text: `Error removing approval of access to ${specificDac.name} to user: ${researcher.displayName}`});
+    }
+  }
+}
+
+const refreshResearchers = async (setResearchers) => {
+  const researcherList = await User.list(USER_ROLES.signingOfficial);
+  // the construction of this list is currently a work-around because our endpoint in the backend
+  // does not currently populate the DAA IDs on the each researcher's libary card
+  const researcherObjectList = await Promise.all(
+    researcherList.map(async (researcher) => {
+      return await User.getById(researcher.userId);
+    })
+  );
+  setResearchers(researcherObjectList);
+}
+
+const DAACell = (
+  rowDac, 
+  researcher,
+  institutionId,
+  daas,
+  refreshResearchers,
+  setResearchers
+) => {
+  const id = researcher && (researcher.userId || researcher.email);
+  const libraryCards = researcher && researcher.libraryCards;
+  const card = libraryCards && libraryCards.find(card => card.institutionId === institutionId);
+  const daaIds = researcher && card && card.daaIds;
+  const filteredDaas = daaIds && daas.filter(daa => daaIds.includes(daa.daaId));
+  const hasDacId = filteredDaas && filteredDaas.some(daa => daa.dacs.some(dac => dac.dacId === rowDac.dacId));
+
+  return {
+    isComponent: true,
+    id,
+    label: 'lc-button',
+    data: (
+      <div>
+        {/* <Checkbox checked={hasDacId}/>  */}
+        <Checkbox checked={hasDacId} onClick={() => handleClick(researcher,rowDac, daas, hasDacId, refreshResearchers, setResearchers)}/> 
```

Comment:
There is a noticeable lag here after clicking since we re-fetch all the users. Should we hold off on this PR until https://github.com/DataBiosphere/consent/pull/2319 is completed? When that is completed, the handler here should be updated to work with the new response from the individual update.

Reply:
If I'm not mistaken, I don't think that PR will solve the lag. I think the ticket/work that will resolve the lag would be [this one](https://broadworkbench.atlassian.net/browse/DCJ-235). In that case, should I pull that in before we move forward with this PR?